src/ProgramSequence.js

```diff
@@ -45,6 +45,16 @@ export default class ProgramSequence {
         return this.program[index];
     }
 
+    hasLoopBlock(): boolean {
+        for (const programBlock of this.program) {
+            const { block } = programBlock;
+            if (block === 'startLoop') {
```

Comment:
I think rather than using destructuring here, we can just use:

    if (programBlock.block === 'startLoop') {